android-components/components/feature/app-links/src/main/java/mozilla/components/feature/app/links/AppLinksInterceptor.kt

```diff
@@ -156,6 +158,7 @@ class AppLinksInterceptor(
             url.startsWith(WWW) -> url.replaceFirst(WWW, "")
             url.startsWith(M) -> url.replaceFirst(M, "")
             url.startsWith(MOBILE) -> url.replaceFirst(MOBILE, "")
+            url.startsWith(MAPS) -> url.replaceFirst(MAPS, "")
```

Comment:
Reading the `doNotIntercept` logic again reminded us of https://github.com/mozilla-mobile/android-components/issues/7614.

Let's add a comment here to describe **why** we're doing this...to not handle the first redirect separately, and so we don't open an additional prompt.